app/controllers/v0/messages_controller.rb

```diff
@@ -67,8 +66,55 @@ def thread
 
     private
 
+    def get_folder(folder_id)
+      folders = client.get_folder(folder_id)
+
+      raise VA::API::Common::Exceptions::RecordNotFound, folder_id unless folders.present?
```

Comment:
can there be multiple folder_ids? if this is a collection action you might not want RecordNotFound. RecordNotFound could work though... just thinking its more suited for member action that is not found maybe.